### PR TITLE
Implement format option

### LINKED ISSUE #9: Implement --format option
(Copied issue from original repository)

```
geekhours course list [--format=xxx]
```

Output in json/csv format.

Add usage to README.md #6

### PR DESCRIPTION
**Pull request by YukieK**
Add some options to format records and write it to a file.

`--format` and `-f` to format records which is output `geekhours course list` and `geekhours done lisrt` commands to CSV or JSON-formatted strings and write to a file.
`--output` and `-o` to specify a file path to save above data.

This PR closes #9.

### CODE REVIEW

File: bin/geekhours [hunk 1]
```diff
@@ -18,9 +18,34 @@ def initialize():
 
 # Handlers for course_subparser
 def list_course(args: str, db_path: str):
-    """ geekhours course list """
+    """ geekhours course list
+
+    Options:
+        --format, -f: Format records to CSV or JSON-formatted strings
```
Comment (mitsutaka) [inline bin/geekhours]: `--format, -f csv|json`

"comma separated CSV or JSON"

File: bin/geekhours [hunk 2]
```diff
@@ -37,9 +62,34 @@ def remove_course(args: str, db_path: str):
 
 # Handlers for done_subparser
 def list_done(args: str, db_path: str):
-    """ geekhours done list """
+    """ geekhours done list
+
+    Options:
+        --format, -f: Format records to CSV or JSON-formatted strings
```
Comment (mitsutaka) [inline bin/geekhours]: Ditto

File: geekhours.md [hunk 3]
```diff
@@ -80,6 +80,35 @@ $ geekhours done list
 ]
 ```
 
+## Format records to CSV or JSON-formatted strings and write it to a file
+
+The '--format' and '-f' options can be used to write records to CSV or JSON file.
```
Comment (mitsutaka) [inline geekhours.md]: Ditto

File: geekhours.md [hunk 4]
```diff
@@ -80,6 +80,35 @@ $ geekhours done list
...
+
+`records.csv` or `records.json` will be created depending on the format type.
+
+```
+$ geekhours done list -f|--format csv|json
+```
+
+### Specify the file name
+
+File name can be specified by giving it to `-output` or `--o` options.
```
Comment (mitsutaka) [inline geekhours.md]: `--output`
`-o`

File: geekhours/command.py [hunk 5]
```diff
@@ -33,3 +40,31 @@ def remove_course(self, arg: str):
...
+    def dump_to_csv(records: str, csvfile: str, fields: Tuple):
+        """ dump outputs to csv format file.
+
+        args:
+            records: Target data to dump.
+            csvfile: File path to save.
+            fields: Tuple object of a header row of the records.
+        """
+
+        with open(csvfile, 'w', newline='') as outcsv:
```
Comment (mitsutaka) [inline geekhours/command.py]: It should be added a check if file exists before open(). If exists, the method should return.

File: geekhours/command.py [hunk 6]
```diff
@@ -33,3 +40,31 @@ def remove_course(self, arg: str):
...
+
+    @staticmethod
+    def dump_to_json(records: str, jsonfile: str):
+        """ dump outputs to json-formatted strings and write it to a file.
+
+        args:
+            records: Target data to write.
+            jsonfile: File path to save.
+        """
+        with open(jsonfile, 'w') as outjson:
```
Comment (mitsutaka) [inline geekhours/command.py]: Ditto

File: bin/geekhours (hunk 1 above)
Comment (YukieK) [reply bin/geekhours]: Fixed.

File: bin/geekhours (hunk 2 above)
Comment (YukieK) [reply bin/geekhours]: Fixed.

File: geekhours.md (hunk 3 above)
Comment (YukieK) [reply geekhours.md]: Fixed.

File: geekhours.md (hunk 4 above)
Comment (YukieK) [reply geekhours.md]: Fixed.

File: geekhours/command.py (hunk 5 above)
Comment (YukieK) [reply geekhours/command.py]: Add `raise FileExistsError` if file already exists.

File: geekhours/command.py (hunk 6 above)
Comment (YukieK) [reply geekhours/command.py]: Add raise FileExistsError if file already exists.

File: geekhours.md [hunk 7]
```diff
@@ -80,6 +80,35 @@ $ geekhours done list
...
+$ geekhours done list -f|--format csv|json
+```
+
+### Specify the file name
+
+File name can be specified by giving it to `-o` or `--output` options.
+
+```
+$ geekhours course list -f csv -o [FILE-PATH]
+$ geekhours done list --format json -output [FILE-PATH]
```
Comment (mitsutaka) [inline geekhours.md]: `--output`

Comment (YukieK) [reply geekhours.md]: Fixed.

**Pull request merged**